src/test/java/org/opensearch/neuralsearch/search/HybridTopScoreDocCollectorTests.java

```diff
@@ -564,4 +564,64 @@ public void testTotalHitsCountValidation_whenTotalHitsCollectedAtTopLevelInColle
         reader.close();
         directory.close();
     }
+
+    @SneakyThrows
+    public void testMaxScoreValidation_whenMaxScoreCollectedAtTopLevelInCollector_thenSuccessful() {
+        final Directory directory = newDirectory();
+        final IndexWriter w = new IndexWriter(directory, newIndexWriterConfig(new MockAnalyzer(random())));
+        FieldType ft = new FieldType(TextField.TYPE_NOT_STORED);
+        ft.setIndexOptions(random().nextBoolean() ? IndexOptions.DOCS : IndexOptions.DOCS_AND_FREQS);
+        ft.setOmitNorms(random().nextBoolean());
+        ft.freeze();
+
+        w.addDocument(getDocument(TEXT_FIELD_NAME, DOC_ID_1, FIELD_1_VALUE, ft));
+        w.addDocument(getDocument(TEXT_FIELD_NAME, DOC_ID_2, FIELD_2_VALUE, ft));
+        w.addDocument(getDocument(TEXT_FIELD_NAME, DOC_ID_3, FIELD_3_VALUE, ft));
+        w.addDocument(getDocument(TEXT_FIELD_NAME, DOC_ID_4, FIELD_4_VALUE, ft));
+        w.commit();
+
+        DirectoryReader reader = DirectoryReader.open(w);
+
+        LeafReaderContext leafReaderContext = reader.getContext().leaves().get(0);
+
+        HybridTopScoreDocCollector hybridTopScoreDocCollector = new HybridTopScoreDocCollector(
+            NUM_HITS,
+            new HitsThresholdChecker(Integer.MAX_VALUE)
+        );
+        LeafCollector leafCollector = hybridTopScoreDocCollector.getLeafCollector(leafReaderContext);
+        assertNotNull(leafCollector);
+
+        Weight weight = mock(Weight.class);
+        int[] docIdsForQuery1 = new int[] { DOC_ID_1, DOC_ID_2 };
+        Arrays.sort(docIdsForQuery1);
+        int[] docIdsForQuery2 = new int[] { DOC_ID_3, DOC_ID_4 };
+        Arrays.sort(docIdsForQuery2);
+        final List<Float> scores = Stream.generate(() -> random().nextFloat()).limit(NUM_DOCS).collect(Collectors.toList());
+        HybridQueryScorer hybridQueryScorer = new HybridQueryScorer(
+            weight,
+            Arrays.asList(
+                scorer(docIdsForQuery1, scores, fakeWeight(new MatchAllDocsQuery())),
+                scorer(docIdsForQuery2, scores, fakeWeight(new MatchAllDocsQuery()))
+            )
+        );
+
+        leafCollector.setScorer(hybridQueryScorer);
+        DocIdSetIterator iterator = hybridQueryScorer.iterator();
+        int nextDoc = iterator.nextDoc();
+        while (nextDoc != NO_MORE_DOCS) {
+            leafCollector.collect(nextDoc);
+            nextDoc = iterator.nextDoc();
+        }
+
+        List<TopDocs> topDocs = hybridTopScoreDocCollector.topDocs();
+        float maxScore = topDocs.stream()
+            .map(docs -> docs.scoreDocs.length == 0 ? new ScoreDoc(-1, 0.0f) : docs.scoreDocs[0])
+            .map(scoreDoc -> scoreDoc.score)
+            .max(Float::compare)
+            .get();
+        assertEquals(maxScore, hybridTopScoreDocCollector.getMaxScore(), 0.0f);
```

Comment:
seems you are depending on objects that are not part of testing. If those dependencies introduce any bug, this test will fail not because of what you want to test. Hence, either mock those external dependencies or convert to integration test if you feel you need to cover the complete flow.

Reply:
I am removing this unit test and will add a integ test instead. It will also address @shatejas and @VijayanB concerns.